browser/src/control/Parts.js

```diff
@@ -278,6 +278,11 @@ L.Map.include({
 	},
 
 	insertPage: function(nPos) {
+		// Make sure we exit cell edit mode.
+		if (this._docLayer.insertMode) {
```

Comment:
This applies only to Calc, but this Part.js is used also in Impress/Draw.
Could you move it below inside the if which checks if it is a spreadsheet?

Also question: couldn't be  that done in the core directly in handling of `.uno:Insert` command? I think it would be better, as there you have "real current state", no race could occur

Reply:
I agree Szymon, I'll handle this in core. Thanks